controllers/multiclusterobservability/multiclusterobservability_controller.go

```diff
@@ -127,11 +131,38 @@ func (r *MultiClusterObservabilityReconciler) Reconcile(ctx context.Context, req
 		return ctrl.Result{}, err
 	}
 
-	//read image manifest configmap to be used to replace the image for each component.
-	if _, err = config.ReadImageManifestConfigMap(r.Client); err != nil {
+	mchCrdExists, err := util.CheckCRDExist(r.CrdClient, config.MCHCrdName)
+	if err != nil {
 		return ctrl.Result{}, err
 	}
 
+	if mchCrdExists {
+		mchList := &mchv1.MultiClusterHubList{}
+		podNamespace, found := os.LookupEnv("POD_NAMESPACE")
+		if !found {
+			podNamespace = config.GetDefaultMCONamespace()
+		}
+		mchistOpts := []client.ListOption{
+			client.InNamespace(podNamespace),
+		}
+		err := r.Client.List(context.TODO(), mchList, mchistOpts...)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		// normally there should only one MCH CR in the cluster
+		if len(mchList.Items) == 1 {
+			mch := mchList.Items[0]
+			mchVer := mch.Status.CurrentVersion
```

Comment:
We need to ensure `CurrentVersion` is equal to `DesiredVersion` then start reconciling. otherwise, just return. like what we support `pause`.

Reply:
done.